woocommerce/woocommerce-android
# Product Creation with AI: Analytics for product creation with AI

Closes: #9843 

Please don't merge until target branch is `trunk`



### Description


Adds the following event trackings:


| Event Name | Trigger | Properties |
|------------|---------|------------|
| *_product_creation_ai_entry_point_displayed | When the action sheet with the option to creation product with AI is displayed. |  |
| *_product_creation_ai_entry_point_tapped | When the option to create product with AI is tapped. |  |
| *_product_creation_ai_product_name_continue_button_tapped | When the Continue CTA on the product name screen is tapped. |  |
| *_product_creation_ai_tone_selected | When a tone is selected for product creation AI. | value: casual \| flowery \| convincing \| formal |
| *_product_creation_ai_generate_details_tapped | When the Generate product details is tapped on the About your product screen. | is_first_attempt: true \| false |
| *_product_creation_ai_generate_product_details_success | When generating product details with AI succeeds. |  |
| *_product_creation_ai_generate_product_details_failed | When generating product details with AI fails. | Error detail properties |
| *_product_creation_ai_save_as_draft_button_tapped | When the Save as draft button on the Preview screen is tapped. |  |
| *_product_creation_ai_save_as_draft_success | When saving an AI-generated product succeeds. |  |
| *_product_creation_ai_save_as_draft_failed | When saving an AI-generated product fails. | Error detail properties |

Extras:
- it also adds `is_ai_content` property (true/false) to the `product_detail_update_button_tapped` event, to mark that content published from this flow has AI content in it.
- also adds `ai_identify_language_success` and `ai_idenfity_language_failed` with `source` value `product_creation`.

### Testing instructions


0. Check the events in logcat:
1. Tap "Add your first product" from onboarding list, check for `product_creation_ai_entry_point_displayed`
2. Tap the product with AI option, check for `product_creation_ai_entry_point_tapped`
3. Enter a name on the next screen, tap Continue, check for `product_creation_ai_product_name_continue_button_tapped`
4. Select a tone on the next screen, check for `product_creation_ai_tone_selected` with the right value
5. Tap "Create product details" on the screen, check for `product_creation_ai_generate_details_tapped` with `is_first_attempt` set to `true`
6. Wait until generation is done, check for `product_creation_ai_generate_product_details_success`
7. Go back, then tap "Create product details" again. check for `product_creation_ai_generate_details_tapped` with `is_first_attempt` set to `false`
8. Tap "Save as draft" at the top, check for `product_creation_ai_save_as_draft_success` once product details screen is loaded.

**Error cases:**
Try re-doing step 6 and step 8 but trigger error (turn off internet for 6).
